### DISCUSSION TITLE
steering committee opinion needed: deprecate "url" config property

### DISCUSSION THREAD

**Discussion by vobu** [ui5-community/wdi5#362] [Ideas]
Hi steering committee @ArnaudBuchholz @dominikfeininger @nicoschoenteich @monavari-lebrecht @hmanchev @aiopa @SebastianWolf-SAP,

your opinion is needed on this suggestion from @Siolto to deprecate the `url` property of the `wdi5` config object in favor of the standard `wdio` `baseUrl` property : https://github.com/ui5-community/wdi5/pull/360
Simon explains his rationale here: https://github.com/ui5-community/wdi5/pull/360#issuecomment-1294854109

> - I often get the feedback that it is quite confusing for people to understand the sense of the additional url property when just passing the whole path to the baseUrl also works perfectly fine. Especially when testing applications in the Launchpad where we don't have an .html file, people get confused.
>  - When you want to dynamically pass the url to the deployed application via cli like wdio run --baseUrl=mydeployedApplication it is almost necessary to leave the url with an empty string and pass the full path in the baseUrl property or use environment variables.
>  - If we don't have a mandatory wdi5 property in the config anymore, the guided installation of wdio also installs and configures everything we need to directly run our tests.

Please give his idea some thought and comment here.
Then - if necessary - we'll cast a vote.

Thx!

**Comment by hmanchev** [comment]
Hi colleagues, 
I agree with @Siolto. This property can be confusing, I also got similar feedback and I can't see any reason to have both url and baseurl. 

BR,
Hristo

**Comment by aiopa** [comment]
Hello all.

Yes, i agree with @hmanchev. If there is no need for a second prop it is better to remove it. I my self got a bit confused at first using the lib.

If we need the base url for some fallback code, i suggest we say something like; baseUrl = url

BR,
Nicholas

**Comment by SebastianWolf-SAP** [comment]
I have to admit I struggle a bit like @vobu did [in his comment](https://github.com/ui5-community/wdi5/pull/360#issuecomment-1294655372). But we should listen to feedback and if 
a) many people are confused because of the current situation and
b) we don't lose any configuration capabilities because of the proposed change
there's a clear indication that it should be done (of course together with a nice deprecation message and documentation).

**Comment by vobu** [comment]
so the majority so far seems in favor of dropping the `wdi5`-specifc `url` config property.
Any opinion from @nicoschoenteich, @monavari-lebrecht  and @ArnaudBuchholz?

**Comment by monavari-lebrecht** [comment]
I follow your arguments and by myself never used it as well. So, plus 1 for deprecation.

**Comment by vobu** [comment]
consensus amongst the committee is in favor of the deprecation then → we'll allow for #360 to get into the 1.0 release!
